homeassistant/components/homekit_controller/connection.py

```diff
@@ -162,6 +166,59 @@ async def async_setup(self):
 
         return True
 
+    async def async_create_devices(self):
+        """
+        Build device registry entries for all accessories paired with the bridge.
+
+        This is done as well as by the entities for 2 reasons. First, the bridge
+        might not have any entities attached to it. Secondly there are stateless
+        entities like doorbells and remote controls.
+        """
+        device_registry = await self.hass.helpers.device_registry.async_get_registry()
+
+        devices = {}
+
+        for accessory in self.entity_map.accessories:
+            info = accessory.services.first(
+                service_type=ServicesTypes.ACCESSORY_INFORMATION,
+            )
+
+            device_info = {
+                "identifiers": {
+                    (
+                        DOMAIN,
+                        "serial-number",
+                        info.value(CharacteristicsTypes.SERIAL_NUMBER),
```

Comment:
I know this is required in theory, but is it possible that the serial number is empty?

Reply:
This is just replicating the existing behaviour for now, so I don't think i'd want to change it in this PR anyway. But to fully answer your question: That has actually happened with homebridge (and actually a 2nd time with homebridge with 0000000), but I don't think we've seen it for a "real" commerical HomeKit implementation. If i was adding homekit_controller from scratch I would use probably use the pairing id and aid here. Unfortunately the serial number is already part of the unique id for an entity, so we don't currently support devices that don't have unique serial numbers anyway.

I'm not sure how to go about changing Entity.unique_id without a breaking change, and with no "real" implementation needing it i've put it waaay down my todo list.